trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/rest/PresignController.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.server.rest;
+
+import com.google.inject.Inject;
+import io.trino.aws.proxy.spi.credentials.Credentials;
+import io.trino.aws.proxy.spi.rest.ParsedS3Request;
+import io.trino.aws.proxy.spi.signing.SigningController;
+import io.trino.aws.proxy.spi.signing.SigningMetadata;
+
+import java.net.URI;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static java.util.Objects.requireNonNull;
+import static java.util.function.Function.identity;
+
+public class PresignController
+{
+    private final SigningController signingController;
+    private final Duration presignUrlDuration;
+
+    @Inject
+    public PresignController(SigningController signingController, TrinoS3ProxyConfig trinoS3ProxyConfig)
+    {
+        this.signingController = requireNonNull(signingController, "signingController is null");
+
+        presignUrlDuration = trinoS3ProxyConfig.getPresignedUrlsDuration().toJavaTime();
+    }
+
+    public Map<String, URI> build(SigningMetadata signingMetadata, ParsedS3Request request, Instant targetRequestTimestamp, URI remoteUri)
+    {
+        Optional<Instant> signatureExpiry = Optional.of(Instant.now().plusMillis(presignUrlDuration.toMillis()));
+
+        return Stream.of("GET", "PUT")
```

Comment:
The problem for multipart is that we don't know how many parts they will have. We could generate a pre-defined number but that seems kludgey. It's not much of a burden to get the pre-signed URLs for each multi-part upload part. I have that all working.

Reply:
We can add `POST` and `DELETE`, though `DELETE` doesn't seem necessary as the point of these pre-signed URLs is to minimize I/O on the proxy; or do you have a different use case?

Reply:
I was trying to make it so that users can choose to avoid going through the proxy altogether if they wish, making the proxy more of a presigned URL generator rather than a proxy. While I don't have a usecase for that in mind, I thought it could come in handy in cases where someone may want to run the proxy in a segmented network without even giving it access to the real S3 / to the internet,

I guess generating presigned upload URLs gets tricky, and if we ever want to do that we could create a custom endpoint that gives you a list of URLs given a desired upload part count.

As for DELETE, I believe it is a simple one and it gets us closer in the direction I mentioned above, but as I said, I don't have a specific use case - this is just speculation

Reply:
I've added all the normal verbs as well as security checks.